### PR TITLE
OOP school library

### PR DESCRIPTION
**Pull request by bigmosi**
Instance vars: @id, @name, and @Age.
Constructor with name, age, and parent_permission as parameters. name and parent_permission are optional and have default values of "Unknown" and true.
Getters for @id, @name, and @Age.
Setters for @name and @Age.
Private method is_of_age? that returns true if @Age is greater or equal to 18 and false otherwise.
Public method can_use_services? that returns true if the person is of age or if they have permission from parents.
Created class Student with the following:
Inherits from Person.
Constructor extends parent's constructor by adding @classroom and a parameter for it.
Method play_hooky that returns "¯(ツ)/¯".
Created class Teacher with the following:
Inherits from Person.
Constructor extends parent's constructor by adding https://github.com/specialization and a parameter for it.
Override can_use_services? so it always returns true.

### CODE REVIEW

File: teacher.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+require_relative 'person'
+# Class Teacher inherits from Person
+class Teachear < Person
+  def initialize(specialization, age)
```
Comment (zieeco) [inline teacher.rb]: - Hey, Please, kindly add the name attribute as a person either [Teacher or Student] should and must have a name. please do add the name attribute.
- Also, do add the parrent_permission to be sure they are not underage to access the resources.

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    super(age, parent_permission: true)
```
Comment (zieeco) [inline teacher.rb]: - Also, add the name property here also to ensure each person has a name.

File: README.md [hunk 3]
```diff
@@ -1 +1,43 @@
...
+- GitHub: [divinecharlotte](https://github.com/divinecharlotte)
+- LinkedIn: [LinkedIn](https://www.linkedin.com/in/charlotte-divine-dusenge/)
+
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the issues page.
+
+Show your support
+Give a star if you like this project!
```
Comment (zieeco) [inline README.md]: - Please, kindly ensure you've followed the layout format as used in the readme template [here](https://github.com/microverseinc/readme-template) to make your readme more professional.
- Add the License section.
- Update your issues page to link to the issues page of your project on Github.
- Add the Acknowledgement section also to make your readme more professional.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @bigmosi,


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — APPROVED** [review]
Hi @bigmosi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**